flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/ParquetInputFormat.java

```diff
@@ -0,0 +1,191 @@
+		super(path);
+		this.readType = new RowTypeInfo(fieldTypes, fieldNames);
+		this.fieldTypes = readType.getFieldTypes();
+		this.fieldNames = readType.getFieldNames();
+		this.unsplittable = true;
+		this.isStandard = isStandard;
+	}
+
+	@Override
+	public Tuple2<Long, Long> getCurrentState() {
+		return new Tuple2<>(this.lastSyncedBlock, this.recordsReadSinceLastSync);
+	}
+
+	@Override
+	public void open(FileInputSplit split) throws IOException {
+		org.apache.hadoop.conf.Configuration configuration = new org.apache.hadoop.conf.Configuration();
+		InputFile inputFile =
+			HadoopInputFile.fromPath(new org.apache.hadoop.fs.Path(split.getPath().toUri()), configuration);
+		ParquetReadOptions options = ParquetReadOptions.builder().build();
+		ParquetFileReader fileReader = new ParquetFileReader(inputFile, options);
+		MessageType schema = fileReader.getFileMetaData().getSchema();
+		MessageType readSchema = getReadSchema(schema);
+		this.parquetRecordReader = new ParquetRecordReader<>(new RowReadSupport(), readSchema, FilterCompat.NOOP);
+		this.parquetRecordReader.initialize(fileReader, configuration);
+		if (this.recordConsumed == null) {
+			this.recordConsumed = getRuntimeContext().getMetricGroup().counter("parquet-record-consumed");
+		}
+		LOG.debug(String.format("Open ParquetRowInputFormat with FileInputSplit [%s]", split.getPath().toString()));
+	}
+
+	@Override
+	public void reopen(FileInputSplit split, Tuple2<Long, Long> state) throws IOException {
+		Preconditions.checkNotNull(split, "reopen() cannot be called on a null split.");
+		Preconditions.checkNotNull(state, "reopen() cannot be called with a null initial state.");
+
+		try {
+			this.open(split);
+		} finally {
+			if (getCurrentState().f0 != -1) {
+				lastSyncedBlock = state.f0;
+				recordsReadSinceLastSync = state.f1;
+			}
+		}
+
+		if (lastSyncedBlock != -1) {
+			// open and read util the record we were before
+			// the checkpoint and discard the values
+			parquetRecordReader.seek(lastSyncedBlock);
+			for (int i = 0; i < recordsReadSinceLastSync; i++) {
+				// skip the record already processed
+				parquetRecordReader.nextRecord();
+			}
+		}
+	}
+
+	@Override
+	public void close() throws IOException {
+		if (parquetRecordReader != null) {
+			parquetRecordReader.close();
+		}
+	}
+
+	@Override
+	public boolean reachedEnd() throws IOException {
+		return parquetRecordReader.reachEnd();
+	}
+
+	@Override
+	public E nextRecord(E e) throws IOException {
+		if (reachedEnd()) {
+			return null;
+		}
+
+		// Means start to create a new block
+		if (parquetRecordReader.getCurrentBlock() != lastSyncedBlock) {
+			lastSyncedBlock = parquetRecordReader.getCurrentBlock();
+			recordsReadSinceLastSync = 0;
+		}
+
+		if (parquetRecordReader.hasNextRecord()) {
+			recordsReadSinceLastSync++;
+			recordConsumed.inc();
+			return convert(parquetRecordReader.nextRecord());
+		}
+
+		return null;
```

Comment:
This case should not happen. Can you add a log statement?

Reply:
Agree. Added